app/keepers.go

```diff
@@ -162,10 +163,14 @@ func InitStakingKeeper(appCodec codec.Codec, keys map[string]*sdk.KVStoreKey, ke
 	return &stakingK
 }
 
-func initWasmKeeper(encodingConfig axelarParams.EncodingConfig, keys map[string]*sdk.KVStoreKey, keepers *KeeperCache, bApp *bam.BaseApp, appOpts types.AppOptions, wasmOpts []wasm.Option, homePath string) *wasm.Keeper {
-	wasmDir := filepath.Join(homePath, "wasm")
+func initWasmKeeper(encodingConfig axelarParams.EncodingConfig, keys map[string]*sdk.KVStoreKey, keepers *KeeperCache, bApp *bam.BaseApp, appOpts types.AppOptions, wasmOpts []wasm.Option, homePath, wasmDir string) *wasm.Keeper {
 	wasmConfig := mustReadWasmConfig(appOpts)
 
+	if wasmDir == "" {
+		dbDir := cast.ToString(appOpts.Get("db_dir"))
+		wasmDir = filepath.Join(homePath, dbDir, "wasm")
+	}
+
```

Comment:
do this outside of initWasmKeeper, then homePath isn't needed here anymore

Reply:
done